src/pages/TeachersUnite/SaveTheWorldPage.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import Navigation from '../../libs/Navigation/Navigation';
+import ROUTES from '../../ROUTES';
+import ONYXKEYS from '../../ONYXKEYS';
+import styles from '../../styles/styles';
+import themeColors from '../../styles/themes/default';
+import Text from '../../components/Text';
+import MenuItem from '../../components/MenuItem';
+import IllustratedHeaderPageLayout from '../../components/IllustratedHeaderPageLayout';
+import * as LottieAnimations from '../../components/LottieAnimations';
+import useLocalize from '../../hooks/useLocalize';
+
+const propTypes = {
+    /** Object with various information about the user */
+    user: PropTypes.shape({
+        /** Whether or not the user is on a public domain email account or not */
+        isFromPublicDomain: PropTypes.bool,
+    }),
+};
+
+const defaultProps = {
+    user: {},
+};
+
+function SaveTheWorldPage(props) {
+    const {translate} = useLocalize();
+    const isLoggedInEmailPublicDomain = props.user.isFromPublicDomain;
```

Comment:
> we can just create a new one once we already have the list on exp-common

Is there a PR for that? If not I can do it today

Reply:
I realized the list is exactly the same from exp-commons, so let's use it

Reply:
I think we shouldn't focus on `isFromPublicDomain` since @parasharrajat reported it not giving accurate results. Instead, we should stick to the function that checks if the mail belongs to a public domain. That will work out of the box for offline scenarios.